File: surf-api-bukkit/surf-api-bukkit-api/src/main/java/dev/slne/surf/surfapi/bukkit/api/nms/listener/packets/serverbound/RenameItemPacket.java
package dev.slne.surf.surfapi.bukkit.api.nms.listener.packets.serverbound;

import org.jetbrains.annotations.ApiStatus.NonExtendable;

@NonExtendable
public non-sealed interface RenameItemPacket extends NmsServerboundPacket {

  String getNewName();
}

File: surf-api-bukkit/surf-api-bukkit-api/src/main/kotlin/dev/slne/surf/surfapi/bukkit/api/nms/listener/packets/serverbound/RenameItemPacket.kt
package dev.slne.surf.surfapi.bukkit.api.nms.listener.packets.serverbound

import dev.slne.surf.surfapi.bukkit.api.nms.NmsUseWithCaution

@NmsUseWithCaution
interface RenameItemPacket : NmsServerboundPacket {
    val newName: String
}
